src/js/hooks/inboundV2/addItems/useInboundAddItemsForm.jsx

```diff
@@ -0,0 +1,564 @@
+  STOCK_MOVEMENT_UPDATE_INVENTORY_ITEMS,
+  STOCK_MOVEMENT_UPDATE_ITEMS,
+  STOCK_MOVEMENT_UPDATE_STATUS,
+} from 'api/urls';
+import { STOCK_MOVEMENT_URL } from 'consts/applicationUrls';
+import InboundV2Step from 'consts/InboundV2Step';
+import { DateFormat } from 'consts/timeFormat';
+import useInboundAddItemsValidation from 'hooks/inboundV2/addItems/useInboundAddItemsValidation';
+import useQueryParams from 'hooks/useQueryParams';
+import useSpinner from 'hooks/useSpinner';
+import useTranslate from 'hooks/useTranslate';
+import apiClient from 'utils/apiClient';
+
+const useInboundAddItemsForm = ({ next, previous }) => {
+  const [loading, setLoading] = useState(false);
+  const spinner = useSpinner();
+  const { validationSchema } = useInboundAddItemsValidation();
+  const queryParams = useQueryParams();
+  const translate = useTranslate();
+  const defaultValues = useMemo(() => {
+    const values = {
+      currentLineItems: [],
+      sortOrder: 0,
+      values: {
+        lineItems: [{
+          palletName: '',
+          boxName: '',
+          product: undefined,
+          lotNumber: '',
+          expirationDate: '',
+          quantityRequested: undefined,
+          recipient: undefined,
+          inventoryItem: undefined,
+        }],
+      },
+      newItem: false,
+      totalCount: 0,
+      isFirstPageLoaded: false,
+      isPaginated: true,
+    };
+    return values;
+  }, []);
+
+  const {
+    control,
+    getValues,
+    handleSubmit,
+    formState: {
+      errors,
+      isValid,
+    },
+    trigger,
+    setValue,
+  } = useForm({
+    mode: 'onBlur',
+    defaultValues,
+    resolver: zodResolver(validationSchema),
+  });
+
+  const confirmSave = (onConfirm) => {
+    confirmAlert({
+      title: translate('react.stockMovement.message.confirmSave.label', 'Confirm save'),
+      message: translate(
+        'react.stockMovement.confirmSave.message',
+        'Are you sure you want to save? There are some lines with empty or zero quantity, those lines will be deleted.',
+      ),
+      buttons: [
+        {
+          label: translate('react.default.yes.label', 'Yes'),
+          onClick: onConfirm,
+        },
+        { label: translate('react.default.no.label', 'No') },
+      ],
+    });
+  };
+
+  const confirmInventoryItemExpirationDateUpdate = (onConfirm) => {
+    confirmAlert({
+      title: translate('react.stockMovement.message.confirmSave.label', 'Confirm save'),
+      message: translate(
+        'react.stockMovement.confirmExpiryDateUpdate.message',
+        'This will update the expiry date across all depots in the system. Are you sure you want to proceed?',
+      ),
+      buttons: [
+        {
+          label: translate('react.default.yes.label', 'Yes'),
+          onClick: onConfirm,
+        },
+        {
+          label: translate('react.default.no.label', 'No'),
+        },
+      ],
+    });
+  };
+
+  const getLineItemsToBeSaved = (lineItems) => {
```

Comment:
I would like this method to be refactored to be more readable. This always has been one of the methods difficult to understand and fix anything inside it. Thoughts @alannadolny ?